cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceProgramRunner.java

```diff
@@ -228,4 +230,16 @@ public void error(Throwable cause) {
     }
     return controller;
   }
+
+  @Nullable
+  private String getAdapterName(Arguments arguments) {
+    // TODO: Currently this logic is super ugly, should try and pass adapter name as its own separate arg
+    if (arguments.hasOption(ProgramOptionConstants.SCHEDULE_NAME)) {
+      String scheduleName = arguments.getOption(ProgramOptionConstants.SCHEDULE_NAME);
+      if (scheduleName.contains(".")) {
```

Comment:
I see, so schedules don't follow the rules for other ``Id`` classes? In that case, we should probably use a different separator for the schedule name set in ``DefaultAdapterConfigurer``. Since its not easy to get ``adapterName`` separately right now. I am unsure if that TODO will get addressed anytime in the immediate future.